Docs/source/running_cpp/parameters.rst

```diff
@@ -1569,25 +1569,50 @@ Reduced Diagnostics
         total mean energy :math:`E_p / \sum w_i`,
         mean energy of each species.
 
-    * ``FieldEnergy``
-        This type computes the electric and magnetic field energy.
-
-        .. math::
-
-            E_f = \sum [ \varepsilon_0 E^2 / 2 + B^2 / ( 2 \mu_0 ) ] \Delta V
-
-        where
-        :math:`E` is the electric field,
-        :math:`B` is the magnetic field,
-        :math:`\varepsilon_0` is the vacuum permittivity,
-        :math:`\mu_0` is the vacuum permeability,
-        :math:`\Delta V` is the cell volume (or area for 2D),
-        the sum is over all cells.
-
-        The output columns are
-        total field energy :math:`E_f`,
-        :math:`E` field energy,
-        :math:`B` field energy, at mesh refinement levels from 0 to :math:`n`.
+    * ``FieldReduced``
+        This type computes values related to the electric and magnetic fields.
+        The list of values to be written must be specified in ``<reduced_diags_name>.field_type``,
```

Comment:
Looking at the input again, `_type` are usually not ideal names to describe things (like `something_action`). In this case, field_type could be understood as `E` or `B` as well.

What do you think about calling those `<diag>.field_reduction` or as in the code `<diag>.field_reduced = <FieldEnergy | MaxField>`?

Reply:
Yes I agree that `field_type` is not a great name, thanks for noting that.
Maybe we can share a common syntax for the field and particle reduced diags? (for now we have `ParticleEnergy` as a reduced diag type but #1316 is also adding `ParticleExtrema`). In which case, this would be a good time to decide on the syntax.

Does something like what is below looks good to you? (maybe `output_quantities` is a bit more explicit than `field_reduced`)
```
warpx.reduced_diag_names = reduced_part reduced_field   ## Current syntax
reduced_part.type = ParticleReduced  ## (or just Particle)
reduced_part.output_quantities = Energy Extrema ## (we can also add Number which could be useful)
reduced_field.type = FieldReduced ## (or just Field)
reduced_field.output_quantities = Energy Maximum
```

@Yin-YinjianZhao What do you think about this?

Reply:
@NeilZaim @ax3l 

I thought about it at the beginning when I started `ParticleExtrema`, but one thing is `ParticleExtrema` contains not only maximum but also minimum values, however, right now this PR only considers maximum fields, right?

Another thing is I kind of prefer many small and clean reduced diagnostics types, rather than a big complicated type that includes many features, since the original purpose of reduced diagnostic is to make it simpler and easier for users, than the full plotfiles, also make it easier for developer to follow and create new ones.

In addition, the name `ParticleReduced` can also mean other types of diagnostics, for example we can even put `BeamRelevant` into it, but I guess we don't want to put everything related to particles into it.

Therefore, my opinion is that, the best solution would be to keep `ParticleEnergy` and `FieldEnergy`, and create new `ParticleExtrema` and `FieldExtrema`, in which `FieldExtrema` also computes both maximum and minimum values.

What do you think? I'd let Axel @ax3l to decide. I'm okay with either way.

Reply:
@NeilZaim and @Yin-YinjianZhao, thanks for synchronizing on this, great idea.

> Another thing is I kind of prefer many small and clean reduced diagnostics types, rather than a big complicated type that includes many features, since the original purpose of reduced diagnostic is to make it simpler and easier for users, than the full plotfiles, also make it easier for developer to follow and create new ones.

Fully agreed. Let's rather write multiple small ones with various types that do one thing well than a few complicated ones. This will keep things simple and we can iterate faster.

> Therefore, my opinion is that, the best solution would be to keep `ParticleEnergy` and `FieldEnergy`, and create new `ParticleExtrema` and `FieldExtrema`, in which `FieldExtrema` also computes both maximum and minimum values.

I think this sounds very reasonable. We can still name the field one "FieldMaximum" since we probably know we are not interested in the minimum anyway and don't want to compute it.

```
warpx.reduced_diag_names = particle_energy_all particle_energy_fast_rear particle_extrema particle_beam_properties field_energy_all field_energy_long field_maximum

particle_energy_all.type = ParticleEnergy

particle_energy_fast_rear.type = ParticleEnergy
particle_energy_fast_rear.filter_function = (uz > 5.0) and (z > 10.0e-6)

particle_extrema.type = ParticleExtrema

particle_beam_properties.type = BeamRelevant
particle_beam_properties.species = e

field_energy_all.type = FieldEnergy
field_energy_long.type = FieldEnergy
field_energy_long.fields = Ez Bx By  # potentially useful to select which ones to mapreduce

field_maximum.type = FieldMaximum
```

I think prior to reduced diagnostics, we used keywords (right-hand side) that are `noun-noun2` (e.g. `particle-energy`). That is also a bit inconsistent now, but let's leave it with that. I documented a few more things we need to unify in #1346

What @Yin-YinjianZhao proposes would avoid another sub option for the selection of diagnostics, which I think is easier to wrap the head around than a type and sub-type of a diagnostics. Keeping the types flat will enable us to continue to add new sub-options when we need them and extend faster.

Very likely sub-options are filters (spatially selected for fields; by properties for particles), selections of field components, etc.

@NeilZaim would that work for you to split this into two reduced diagnostics? Sorry for the back and forth and thank you for synchronizing this.

Reply:
Thanks for the feedback @Yin-YinjianZhao and @ax3l .

In the end I agree with you that having multiple small reduced diagnostics is better for the user (having a txt file with 15+ columns of 10-digits numbers is not very readable) so I will revert this back to 2 separate reduced diagnostics.

I also agree with @ax3l that FieldMaximum is sufficient (we don't need the extrema). It might be useful in some specific cases to have the maximum value of a field component in both directions but I think for the most part we're mainly interested in maximum absolute values so this is already a good starting point.

Reply:
@NeilZaim Thank you! Sorry for the extra work of splitting!